File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/AttendanceRepository.java
package com.example.schoolsite.workWithDatabase.repo;

import com.example.schoolsite.entity.Attendance;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface AttendanceRepository extends JpaRepository<Attendance, Long> {
    boolean existsByClassIDAndLessonIDAndPupilID(Long classId, Long lessonId, Long pupilId);
    List<Attendance> findAllByPupilID(Long pupilId);
}

File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/AttendanceRepository.kt
package com.example.schoolsite.workWithDatabase.repo

import com.example.schoolsite.entity.Attendance
import org.springframework.data.jpa.repository.JpaRepository
import java.util.List
import org.springframework.stereotype.Repository

@Repository
interface AttendanceRepository : JpaRepository<Attendance?, Long?> {
    fun existsByClassIDAndLessonIDAndPupilID(classId: Long?, lessonId: Long?, pupilId: Long?): Boolean
    fun findAllByPupilID(pupilId: Long?): List<Attendance?>?
}